src/main.rs

```diff
@@ -1,100 +1,119 @@
-                        continue;
-                    }
-                    // Get standard account address. If multisig:
-                    let account_address_hex = if multisig {
-                        println!("Multisig account address: 0x{}", search_str);
-                        // Re-derive account address bytes here rather than cloning during search.
-                        hex::encode(auth_key_bytes_vec(&private_key))
-                    } else {
-                        search_str // If not multisig, standard account address is from search.
-                    };
-                    println!("Standard account address: 0x{}", account_address_hex);
-                    println!(
-                        "Private key:              0x{}\n",
-                        hex::encode(private_key.to_bytes())
-                    );
-                }
-            }
-        });
+
+    if args.prefix.len() % 2 == 1 {
+        hex::decode(format!("{}0", &args.prefix))?;
+    } else {
+        hex::decode(&args.prefix)?;
     }
-    // Run infinite loop to prevent main thread from returning and closing search threads.
+
+    Ok(args)
+}
+
+fn generate_key(
+    prefix: String,
+    multisig: bool,
+    res_tx: channel::Sender<(String, String)>,
+    exit_rx: channel::Receiver<()>,
+) -> Result<()> {
+    let mut key_generator = KeyGen::from_os_rng();
+    let prefix_bytes = hex::decode(&prefix)?;
     loop {
-        thread::sleep(Duration::from_secs(u64::MAX)); // Sleep to preserve resources.
+        if exit_rx.try_recv().is_ok() {
+            return Ok(());
+        }
+
+        let private_key = key_generator.generate_ed25519_private_key();
+        let account_address_bytes = auth_key_bytes_vec(&private_key);
+        let search_bytes = if multisig {
+            create_multisig_account_address(AccountAddress::from_bytes(account_address_bytes)?, 0)
+                .to_vec()
+        } else {
+            // If standard account, check against account address.
+            account_address_bytes
+        };
+
+        // If a match found, print diagnostics then keep searching.
+        if search_bytes.starts_with(&prefix_bytes) {
+            let search_str = hex::encode(search_bytes);
+            if !search_str.starts_with(&prefix) {
+                continue;
+            }
+            // Get standard account address. If multisig:
+            let account_address_hex = if multisig {
+                println!("Multisig account address: 0x{}", search_str);
+                // Re-derive account address bytes here rather than cloning during search.
+                hex::encode(auth_key_bytes_vec(&private_key))
+            } else {
+                // If not multisig, standard account address is from search.
+                search_str
+            };
+
+            res_tx.send((
+                account_address_hex.clone(),
+                hex::encode(private_key.to_bytes()),
+            ))?;
```

Comment:
This results in garbled messages for fast searches with fewer characters.

For example

```
Multisig account address: 0xc0debb39a954cab8b2ac79b2a0ba41c6e8f410e867cde088a468fe25be2d738a
Multisig account address: 0xc0de34d76bc2c672628de816191b1d4523830f726caadba4bce32ca7db0608e0
Standard account address: 0xc371536c6a5df49ca986ba2bd67c8157424f39ddd40a39145c3e3659c10efcd5
Private key:              0x5f55c2617ce2b28d037416d25e9e466f38d82d0cac1d1a47ffb563ab1975852d

Standard account address: 0x41dd6de12e4bcd21b813c919f44921c6a676d7df9b07f45fd17265def44b8145
Private key:              0xdffbf8ea653f293e1bdb2bb4b8ca05a4c844a05135b4e9c5b903d7233e2147e8
```